app/Jobs/ImportTurboVotePosts.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace Rogue\Jobs;
+
+use Carbon\Carbon;
+use League\Csv\Reader;
+use Rogue\Models\Post;
+use Rogue\Models\Signup;
+use Illuminate\Bus\Queueable;
+use Rogue\Services\Three\PostService;
+use Illuminate\Support\Facades\Storage;
+use Rogue\Services\Three\SignupService;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class ImportTurboVotePosts implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    /**
+     * The path to the stored csv.
+     *
+     * @var string
+     */
+    protected $filepath;
+
+    /**
+     * The role of the authenticated user that kicked off the request.
+     *
+     * @var object
+     */
+    protected $authenticatedUserRole;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($filepath, $authenticatedUserRole)
+    {
+        $this->filepath = $filepath;
+
+        $this->authenticatedUserRole = $authenticatedUserRole;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(SignupService $signupService, PostService $postService)
+    {
+        // Get the file from storage based on path.
+        $file = Storage::get($this->filepath);
+
+        // Read in the file and get the records.
+        $csv = Reader::createFromString($file);
+        $csv->setHeaderOffset(0);
+        $records = $csv->getRecords();
+
+        foreach ($records as $record) {
+            info('Importing record ' . $record['id'], ['record' => $record]);
```

Comment:
@sbsmith86 Do we have any sense of how large we expect these CSVs to be? I want to be mindful of not overloading the logs, but I also recognize that we may want detailed records so we can verify that each signup got created.

Reply:
I will double check with @ngjo but I don't think we will have a good idea about this until it is out in the wild. I prefer to have more logs and cut back if we need to since I want to make sure we can track how these run.

Reply:
Sounds like a good game plan to me!